app/services/form_response.rb

```diff
@@ -38,6 +38,11 @@ def first_error_message
     Array(message_or_messages).first
   end
 
+  def failure_reasons
+    return if errors.blank?
+    errors.map{|k, v| "#{k.to_s} #{[v].flatten.join(' ')}" }.join(", ")
+  end
```

Comment:
I think what we really want is something more similar to `hash[:error_details]` like above in this file, because when provided, those should be the slugs that are locale-independent